net/levin/src/lib.rs

```diff
@@ -128,14 +132,14 @@ impl MessageType {
 
     /// Returns the `MessageType` given the flags and have_to_return_data fields
     pub fn from_flags_and_have_to_return(
-        flags: header::Flags,
+        flags: Flags,
         have_to_return: bool,
     ) -> Result<Self, BucketError> {
-        if flags.is_request() && have_to_return {
+        if flags.contains(Flags::REQUEST) && have_to_return {
             Ok(MessageType::Request)
-        } else if flags.is_request() {
+        } else if flags.contains(Flags::REQUEST) {
```

Comment:
Compiler probably does it anyway but you could store/reuse the `flags.contains(Flags::REQUEST)`.